nautobot/ipam/views.py

```diff
@@ -661,6 +677,127 @@ def post(self, request):
         )
 
 
+class IPAddressMergeView(view_mixins.GetReturnURLMixin, view_mixins.ObjectPermissionRequiredMixin, View):
+    queryset = IPAddress.objects.all()
+    template_name = "ipam/ipaddress_merge.html"
+
+    def get_required_permission(self):
+        return get_permission_for_model(self.queryset.model, "change")
+
+    def find_duplicate_ips(self, request, merged_attributes=None):
+        if merged_attributes:
+            host_values = (
+                self.queryset.filter(host__gt=merged_attributes.get("host"))
+                .values("host")
+                .order_by("host")
+                .annotate(count=models.Count("host"))
+                .filter(count__gt=1)
+            )
+        else:
+            host_values = (
+                self.queryset.values("host").order_by("host").annotate(count=models.Count("host")).filter(count__gt=1)
+            )
+        if host_values:
+            item = host_values[0]
+            queryset = self.queryset.filter(host__in=[item["host"]])
+            return render(
+                request=request,
+                template_name=self.template_name,
+                context={
+                    "queryset": queryset,
+                    "return_url": self.get_return_url(request),
+                },
+            )
+        else:
+            msg = "No additional duplicate IPs found."
+            messages.warning(request, msg)
+            return redirect(self.get_return_url(request))
+
+    def get(self, request):
+        return self.find_duplicate_ips(request)
+
+    def post(self, request):
+        logger = logging.getLogger(__name__)
+        collapsed_ips = IPAddress.objects.filter(pk__in=request.POST.getlist("pk"))
+        merged_attributes = request.POST
+        # Check if there are at least two IP addresses for us to merge
+        if len(collapsed_ips) < 2:
+            msg = "Invalid Operation, please select at least two IP Addresses to merge."
+            messages.error(request, mark_safe(msg))
+            return redirect(self.get_return_url(request))
+        if "_skip" not in request.POST:
+            with cache.lock("ipaddress_merge", blocking_timeout=15, timeout=settings.REDIS_LOCK_TIMEOUT):
+                with transaction.atomic():
+                    namespace = Namespace.objects.get(pk=merged_attributes.get("namespace"))
+                    status = Status.objects.get(pk=merged_attributes.get("status"))
+                    if merged_attributes.get("tenant"):
+                        tenant = Tenant.objects.get(pk=merged_attributes.get("tenant"))
+                    else:
+                        tenant = None
```

Comment:
In the interest of reducing code duplication, the following is functionally equivalent for the lines where you're getting a single object or defaulting to `None`:

```suggestion
                    tenant = Tenant.objects.filter(pk=merged_attributes.get("tenant")).first()
```

If you call `.first()` on a empty queryset it returns `None`.

Reply:
merged_attributes.get("tenant") could return "" which is not a valid UUID and `filter(pk=merged_attributes.get("tenant")` could throw an error because of that.